components/eam/src/chemistry/modal_aero/modal_aero_convproc.F90

```diff
@@ -2230,163 +2221,195 @@ subroutine ma_activate_convproc_method2(     &
    integer,  intent(in)    :: kactfirst ! k at cloud base
 
 !-----------------------------------------------------------------------
 ! local variables
-   integer  :: l, ll, la, lc, n
-
-   real(r8) :: delact                ! working variable
-   real(r8) :: dt_u_inv              ! 1.0/dt_u
-   real(r8) :: fluxm(ntot_amode)      ! to understand this, see subr activate_modal
-   real(r8) :: fluxn(ntot_amode)      ! to understand this, see subr activate_modal
-   real(r8) :: flux_fullact           ! to understand this, see subr activate_modal
-   real(r8) :: fm(ntot_amode)         ! mass fraction of aerosols activated
-   real(r8) :: fn(ntot_amode)         ! number fraction of aerosols activated
-   real(r8) :: hygro(ntot_amode)      ! current hygroscopicity for int+act
-   real(r8) :: naerosol(ntot_amode)   ! interstitial+activated number conc (#/m3)
-   real(r8) :: sigw                  ! standard deviation of updraft velocity (cm/s)
-   real(r8) :: smax_prescribed       ! prescribed supersaturation for secondary activation (0-1 fraction)
-   real(r8) :: tmpa, tmpb, tmpc      ! working variable
-   real(r8) :: tmp_fact              ! working variable
-   real(r8) :: vaerosol(ntot_amode)   ! int+act volume (m3/m3)
-   real(r8) :: wbar                  ! mean updraft velocity (cm/s)
-   real(r8) :: wdiab                 ! diabatic vertical velocity (cm/s)
-   real(r8) :: wminf, wmaxf          ! limits for integration over updraft spectrum (cm/s)
-
+   integer  :: la, lc, imode, ispec   ! indices
+
+   real(r8) :: dt_u_inv               ! 1.0/dt_u [1/s]
+   real(r8) :: fluxm(ntot_amode)      ! output of activate_modal not used in this subroutine. to understand this, see subr activate_modal
+   real(r8) :: fluxn(ntot_amode)      ! output of activate_modal not used in this subroutine. to understand this, see subr activate_modal
+   real(r8) :: flux_fullact           ! output of activate_modal not used in this subroutine. to understand this, see subr activate_modal
+   real(r8) :: fm(ntot_amode)         ! mass fraction of aerosols activated [fraction]
+   real(r8) :: fn(ntot_amode)         ! number fraction of aerosols activated [fraction] 
+   real(r8) :: hygro(ntot_amode)      ! current hygroscopicity for int+act [unitless]
+   real(r8) :: naerosol(ntot_amode)   ! interstitial+activated number conc [#/m3]
+   real(r8) :: sigw                   ! standard deviation of updraft velocity [m/s]
+   real(r8) :: act_frac               ! activation fraction [fraction]
+   real(r8) :: vaerosol(ntot_amode)   ! int+act volume [m3/m3]
+   real(r8) :: wbar                   ! mean updraft velocity [m/s]
+   real(r8) :: wdiab                  ! diabatic vertical velocity [m/s]
+   real(r8) :: wminf, wmaxf           ! limits for integration over updraft spectrum [m/s]
 
 !-----------------------------------------------------------------------
 
-
-
 ! check f_ent > 0
-   if (f_ent <= 0.0) return
+   if (f_ent <= 0.0_r8) return
```

Comment:
It is a good thing to add `_r8` but sometimes it does change answers and make them NBFB. So just be careful about it.